lib/rules/no-debug.ts

```diff
@@ -83,13 +89,22 @@ export default createTestingLibraryRule<Options, MessageIds>({
       },
       CallExpression(node) {
         const callExpressionIdentifier = getDeepestIdentifierNode(node);
```

Comment:
@timdeschryver This is really weird: `getDeepestIdentifierNode` (and some other utils for finding identifiers) are typed as returning `TSESTree.Identifier | null`. However, TS is assuming they always returning `TSESTree.Identifier` and never `null`:

![CleanShot 2021-04-02 at 12 02 28](https://user-images.githubusercontent.com/2677072/113410556-00b21800-93b4-11eb-963c-8a2c054e2dfe.png)

Then, if I update the function to `return ''` instead of `return null` and update the returned type to `TSESTree.Identifier | string`, TS starts complaining about returned value saved in all these rules might not be an `Identifier`. Is this a bug on TS? Am I missing something else? I don't get why it always assumes `null` won't be returned.

Reply:
I suspect turning on [strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks) in the tsconfig would fix that.

Reply:
Oh wow... TIL...
@epmatsw is right, you'll have to turn the strict mode on.
I would encourage to turn on strict mode though, not just strict null checks.

Reply:
Aah damn! I thought we had `strict` enabled, but anyway I didn't expect those nullable values to be silent. I'm gonna enable strict mode and fix all corresponding issues then. Thanks guys.